### ISSUE TITLE
Agent unable to re-attest with x509pop NodeAttestor

### ISSUE THREAD

**Issue by jpeg** [spiffe/spire#552]
* **Version**: 0.6.1, master@dc91d3041be7e68666d4a262f0aa73873e46bd96
* **Platform**: Redhat 7.4 Docker container
* **Subsystem**: Server

Based on a discussion on the SPIFFE slack, unlike some NodeAttestor plugins, re-attestation with the x509pop plugin is likely safe to allow as long as the private key is secured stored. However attempting to do so currently results in a server-side error.

After upgrading from 0.6.0 to 0.6.1 entries are now being written to the node_resolver_map_entries database table by the x509pop plugin, I think https://github.com/spiffe/spire/pull/516 is the change responsible. When attempting to attest for a second time with the same X.509 certificate it fails with the following error on the server:
```
DEBU[0053] Signing CSR for Agent SVID spiffe://cerner.local/spire/agent/x509pop/1cc955ba8fd2b63e981bef6b6826d924c0b8637b  subsystem_name=node_api
ERRO[0053] UNIQUE constraint failed: node_resolver_map_entries.spiffe_id, node_resolver_map_entries.type, node_resolver_map_entries.value  subsystem_name=node_api
```
The agent then receives an Unknown error and crashes:
```
ERRO[0000] agent crashed: attesting to SPIRE server: rpc error: code = Unknown desc = Error trying to get selectors for baseSpiffeID
```

Steps to repeat:
1. Attest agent configured with x509pop NodeAttestor plugin & memory KeyManager plugin.
1. Restart agent process.
1. Fails on node attestation with above errors.

**Comment by evan2645** [comment]
Thank you very much for this report @jpeg. It is definitely a bug in resolution entry management.

I'll note that there is a `KeyManager "disk"` [plugin](https://github.com/spiffe/spire/blob/master/doc/plugin_agent_keymanager_disk.md) available. This plugin allows the agent to store its keys on disk so that re-attestation is not necessary following agent restart or reboot. I am not sure if this is a suitable workaround for your use case or not, but thought I would share it.

I think the correct behavior here should be to allow attestation to succeed, and/or to detect duplicate entries.

**Comment by azdagron** [comment]
I believe this was fixed as a side-effect of #565.
